pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/nullvalue/NullValueIndexType.java

```diff
@@ -67,14 +67,11 @@ public IndexConfig getDefaultConfig() {
 
   @Override
   public ColumnConfigDeserializer<IndexConfig> createDeserializer() {
-    return IndexConfigDeserializer.fromIndexes("null", getIndexConfigClass())
-        .withFallbackAlternative(
```

Comment:
I am not very familiar with the code. What does this withFallbackAlternative do?

Reply:
It is one of the options on `MergedColumnConfigDeserializer`. Specifically, it uses `MergedColumnConfigDeserializer.OnConflict.PICK_FIRST`, which means that in case a configuration is specified twice, it will pick the first and silently skip the second.

In this case it meant that first it would look for something like:
```js
{
  fieldConfigList: [
    {
       name: "whatever",
       indexes: {
          null: {} // this is what IndexConfigDeserializer.fromIndexes looks for
       }
    }
  }
}
```

And if it is not found, it would look for the alternative which is to read `tableConfig.getIndexingConfig().isNullHandlingEnabled()`. To be more precise, both alternatives will always be read and then merged silently. That means that in case the config is incorrectly specified in one of the branches, we are going to produce a failure.

The new code removes the first part in order to do not look for nullability in `fieldConfigList`, which is desired because it is actually not supported, so we would be generating an extra index that would never be used.